pkg/autodiscovery/providers/cloudfoundry.go

```diff
@@ -0,0 +1,215 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+//
+// +build clusterchecks
+
+package providers
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/cloudfoundry"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/bhmj/jsonslice"
+)
+
+// CloudFoundryConfigProvider implements the Config Provider interface, it should
+// be called periodically and returns templates from Cloud Foundry BBS for AutoConf.
+type CloudFoundryConfigProvider struct {
+	bbsCache      cloudfoundry.BBSCacheI
+	lastCollected time.Time
+}
+
+// NewCloudFoundryConfigProvider instantiates a new CloudFoundryConfigProvider from given config
+func NewCloudFoundryConfigProvider(conf config.ConfigurationProviders) (ConfigProvider, error) {
+	cfp := CloudFoundryConfigProvider{
+		lastCollected: time.Now(),
+	}
+	var err error
```

Comment:
we don't need to create that var, it could be local to the if

Reply:
I don’t think this is doable, because you can’t do `cfp.bbsCache, err := …`, because that would raise compilation error `non-name cfp.bbsCache on the left side of :=`, so the error has to be defined before the `if` and plain `=` has to be used - please correct me if I'm missing something.